### PR TITLE
76: fix warnings and notes from check as much as possible

### LINKED ISSUE #76: Address warnings and notes (as much as possible) from `R CMD check`
On enableR:
The following notes and warnings are generated when `R CMD check` is run against this package.

```
* checking compilation flags used ... NOTE
Compilation used the following non-portable flag(s):
  ‘-Werror=format-security’ ‘-Wp,-D_FORTIFY_SOURCE=2’
  ‘-Wp,-D_GLIBCXX_ASSERTIONS’
* checking compiled code ... WARNING
File ‘mmrm/libs/mmrm.so’:
  Found ‘abort’, possibly from ‘abort’ (C)
    Objects: ‘mmrm.o’, ‘test-covariance.o’, ‘test-runner.o’,
      ‘test-utils.o’, ‘tmb.o’
  Found ‘printf’, possibly from ‘printf’ (C)
    Objects: ‘mmrm.o’, ‘test-covariance.o’, ‘test-runner.o’,
      ‘test-utils.o’, ‘tmb.o’
File ‘mmrm/libs/mmrm.so’:
  Found no calls to: ‘R_registerRoutines’, ‘R_useDynamicSymbols’

Compiled code should not call entry points which might terminate R nor
write to stdout/stderr instead of to the console, nor use Fortran I/O
nor system RNGs.
It is good practice to register native routines and to disable symbol
search.

See ‘Writing portable packages’ in the ‘Writing R Extensions’ manual.
```
To do:
- [x] be able to reproduce this - on enableR?...
- [x] fix it if possible
  - [x] it is not clear where the non-portable flags come from. Possibly from TMB. However this warning does not show up without CRAN flag or on R 4.2 so should be ok.
  - [x] it is not clear where the `abort` and `printf` symbols come from. I checked the C++ code and we don't have any of this or similar. Maximum we have is `Rf_error` which is standard R interface for an error so cannot be the problem. Note that this warning does not occur on R 4.2

On Daniel's Macbook with `R CMD check --as-cran` on the built package tarball:

```
* checking compiled code ... NOTE
Datei ‘mmrm/libs/mmrm.so’:
  Found no calls to: ‘R_registerRoutines’, ‘R_useDynamicSymbols’

It is good practice to register native routines and to disable symbol
search.

See ‘Writing portable packages’ in the ‘Writing R Extensions’ manual.
```
To do: 
- [x] See https://stackoverflow.com/questions/42313373/r-cmd-check-note-found-no-calls-to-r-registerroutines-r-usedynamicsymbols how to fix this.
- [x] fix it
- [x] ongoing work, as it is more complicated than initially thought:
	- [x] see https://github.com/kaskr/adcomp/blob/00046bb67271fa3db266cabd5bd815b58fa0045d/TMB/inst/include/tmb_core.hpp
	- [x] see https://github.com/glmmTMB/glmmTMB/blob/master/glmmTMB/src/init.h

Moreover parallel processes are not allowed in CRAN:
```
 ✖ | 2      47 | fit [0.7s]                                                      
  ────────────────────────────────────────────────────────────────────────────────
  Error (test-fit.R:175:3): mmrm falls back to other optimizers if default does not work
  Error in `.check_ncores(cores)`: 3 simultaneous processes spawned
  Backtrace:
    1. testthat::expect_silent(mmrm(formula, data_small))
         at test-fit.R:175:2
    9. mmrm::mmrm(formula, data_small)
   10. mmrm::refit_multiple_optimizers(fit, n_cores = n_cores, accept_singular = accept_singular)
   13. parallel::mclapply(...)
   14. parallel:::.check_ncores(cores)
  
  Error (test-fit.R:185:3): mmrm fails if no optimizer works
  Error in `.check_ncores(cores)`: 3 simultaneous processes spawned
  Backtrace:
    1. testthat::expect_error(...)
         at test-fit.R:185:2
    7. mmrm::mmrm(formula, data_small, reml = FALSE)
    8. mmrm::refit_multiple_optimizers(fit, n_cores = n_cores, accept_singular = accept_singular)
   11. parallel::mclapply(...)
   12. parallel:::.check_ncores(cores)
  ────────────────────────────────────────────────────────────────────────────────
```
To do:
- [x] disable parallel processing in the 2 tests

### PR DESCRIPTION
**Pull request by danielinteractive**
closes #76

To review:

- [ ] look through code that nothing weird is included
- [ ] checkout from feature branch
- [ ] build source package with `R CMD build mmrm`
- [ ] do `R CMD check --as-cran` on the source package and confirm that no warnings or errors occur

### CODE REVIEW

**Comment by danielinteractive** [comment]
@dinakar29 would it actually be possible to always use `--as-cran` for this package checks?

**Comment by dinakar29** [comment]
> @dinakar29 would it actually be possible to always use `--as-cran` for this package checks?

Yes, we have this feature. Let me add it here.

**Review by dgkf — APPROVED** [review]
Approved. A couple comments, but just code discussion, no changes needed. One compilation note:

When checking, I got this note as well:
```
checking compilation flags used ... NOTE
  Compilation used the following non-portable flag(s):
    ‘-Wdate-time’ ‘-Werror=format-security’ ‘-Wformat’
```

Compiled on a Ubuntu machine within a `rocker/tidyverse` podman container. It sounds like they can be safely ignored, as these are set by the Ubuntu builds of R and not as part of the package (https://github.com/r-hub/rhub/issues/264#issuecomment-490874439).

File: tests/testthat/test-fit.R [hunk 1]
```diff
@@ -182,8 +183,9 @@ test_that("mmrm fails if no optimizer works", {
 
   formula <- FEV1 ~ RACE + SEX + ARMCD * AVISIT + us(AVISIT | USUBJID)
   data_small <- fev_data[1:30, ]
+  # Note: We disable parallel processing here to comply with CRAN checks.
   expect_error(
-    mmrm(formula, data_small, reml = FALSE),
+    mmrm(formula, data_small, reml = FALSE, n_cores = 1L),
```
[the remark this answers is not in the document]
Comment (danielinteractive) [reply tests/testthat/test-fit.R:188]: ok let me add that still here, it is fine to run parallel on CI, just not on CRAN (e.g. it was parallel here before)

**Pull request merged**